tests/client/core/test_utils.js

```diff
@@ -172,56 +180,169 @@ describe('getClientApp', () => {
-      assert.strictEqual(clientSupportsAddons(ua), true, `UA string: ${ua}`);
+  it('is incompatible with Chrome desktop', () => {
+    userAgents.chrome.forEach((userAgent) => {
+      assert.deepEqual(
+        isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
+        { compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX },
+        `UA string: ${userAgent}`);
     });
   });
 
-  it('returns true for Firefox Android', () => {
-    userAgents.firefoxAndroid.forEach((ua) => {
-      assert.strictEqual(clientSupportsAddons(ua), true, `UA string: ${ua}`);
+  it('is compatible with Firefox desktop', () => {
+    userAgents.firefox.forEach((userAgent) => {
+      assert.deepEqual(
+        isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
+        { compatible: true },
+        `UA string: ${userAgent}`);
     });
   });
 
-  it('is case insensitive', () => {
-    const ua = userAgents.firefox[0].toLowerCase();
-    assert.strictEqual(clientSupportsAddons(ua), true, `UA string: ${ua}`);
+  it('is compatible with Firefox Android', () => {
+    userAgents.firefoxAndroid.forEach((userAgent) => {
+      assert.deepEqual(
+        isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
+        { compatible: true },
+        `UA string: ${userAgent}`);
+    });
   });
 
-  it('returns true for Firefox OS', () => {
-    userAgents.firefoxOS.forEach((ua) => {
-      assert.strictEqual(clientSupportsAddons(ua), true, `UA string: ${ua}`);
+  it('is compatible with Firefox OS', () => {
+    userAgents.firefoxOS.forEach((userAgent) => {
+      assert.deepEqual(
+        isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
+        { compatible: true },
+        `UA string: ${userAgent}`);
     });
   });
 
-  it('returns false for Firefox iOS', () => {
-    userAgents.firefoxIOS.forEach((ua) => {
-      assert.strictEqual(clientSupportsAddons(ua), false, `UA string: ${ua}`);
+  it('is incompatible with Firefox iOS', () => {
+    userAgents.firefoxIOS.forEach((userAgent) => {
+      assert.deepEqual(
+        isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
+        { compatible: false, reason: INCOMPATIBLE_FIREFOX_FOR_IOS },
+        `UA string: ${userAgent}`);
     });
   });
 
-  it('returns false for non-string user agent values', () => {
-    assert.strictEqual(clientSupportsAddons(false), false);
+  it('should mark Firefox for iOS as incompatible reason, not version', () => {
+    const userAgentInfo = {
+      browser: { name: 'Firefox', version: '8.0' },
+      os: { name: 'iOS' },
+    };
+    assert.deepEqual(
+      isCompatibleWithUserAgent({ minVersion: '9.0', userAgentInfo }),
+      { compatible: false, reason: INCOMPATIBLE_FIREFOX_FOR_IOS }
+    );
+  });
+
+  it('is incompatible with garbage user agent', () => {
+    const userAgent = '^^**garbage__';
+    assert.deepEqual(
+      isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
```

Comment:
I think it would be more useful to put this test in the reducer since this is really just an exercise for the parsing code.